### PR TITLE
Update instructions for Kustomize 5

### PR DESCRIPTION
**Pull request by kimwnasptd**
Refs https://github.com/kubeflow/manifests/issues/2388

This PR updates the instructions and the example kustomization to work with Kustomize 5. Note that after this PR is merged users won't be able to use earlier versions of kustomize, since we need the `sortOptions` feature of Kustomize 5 https://github.com/kubeflow/manifests/issues/2388#issuecomment-1440073723. This holds for the one-liner install. The manual instructions are not affected and people can use earlier versions of kustomize there.

Next work items, although not critical for the KF 1.7 release, are to update our manifests so that we don't see warnings from Kustomize https://github.com/kubeflow/manifests/issues/2388#issuecomment-1440044713

cc @DomFleischmann @annajung @juliusvonkohout @jbottum 
/cc @kubeflow/wg-manifests-leads

### CODE REVIEW

**Comment by annajung** [comment]
Thanks Kimonas! I couldn't recreate the race condition described in the https://github.com/kubeflow/manifests/issues/1797 with kustomize 5, but can confirm that with sortOptions added, it does create `MutatingWebhookConfiguration` before the necessary resources.

Testing with `kustomize build example`, only difference between kustomize5 with sortOptions enabled and kustomize3 was the order of when PVCs were created. It looks like kustomize3 places PVC at the end right before the ValidatingWebhookConfiguration, so kustomize5 order is preferrable IMO

lgtm (not adding the actual label to allow other folks to review)

there is still ongoing discussion on if it should be included in 1.7 release in https://github.com/kubeflow/manifests/issues/2388 but should not be a blocker for this PR.

File: README.md [hunk 1]
```diff
@@ -438,7 +440,3 @@ For security reasons, we don't want to use the default password for the default
 
 - **Q:** What versions of Istio, Knative, Cert-Manager, Argo, ... are compatible with Kubeflow? \
   **A:** Please refer to each individual component's documentation for a dependency compatibility range. For Istio, Knative, Dex, Cert-Manager and OIDC-AuthService, the versions in `common` are the ones we have validated.
```
Comment (DomFleischmann) [inline README.md:442]: Here is a suggestion for the Kustomize 3.2 QA section.

```suggestion
  **A:** Please refer to each individual component's documentation for a dependency compatibility range. For Istio, Knative, Dex, Cert-Manager and OIDC-AuthService, the versions in `common` are the ones we have validated.
  
- **Q: ** How can I still use Kustomize version 3.2?
  **A: ** The only thing incompatible with Kustomize 3.2 is the `sortOptions` section in the `example/kustomization.yaml` file. I you want to use Kustomize 3.2 just comment out the entire section (`#` character in front of each line) and execute the one-liner. 
```

Comment (kimwnasptd) [reply README.md:442]: @DomFleischmann I pushed a commit with a slightly refactored section for the earlier version of Kustomize. PTAL

**Comment by kimwnasptd** [comment]
Adding a summary here as well from the last community (Tue 28 Feb) and release meeting (Mon 6 March). We decided to move forward with using Kustomize 5 as the default installation tool.

The rationale is that:
1. With feedback until now we have heard from distros that they managed to use Kustomize 5 without any issues
2. There are no (currently known) deprecations that affect our use of kustomize
3. We'd like to move to a newer version of Kustomize, which hasn't been updated for a long time
4. We have documented a clear path on how people can use earlier versions of Kustomize, if they need to

So at this point we have confidence from both Manifests WG and feedback from distros that Kustomize 5 does not have any issues we haven't yet documented. Plus if adopters are not ready for the version switch they can still use an earlier version of kustomize. So let's move forward with this PR and also cherry-pick it to the KF 1.7 release branch

**Pull request merged**